All tests failing due to the config validation are fixed in the new revision [1] barring one test, which is flaky and fails intermittently due to unrelated reasons and passes separately [2]

[1]
```
2256 tests completed, 1 failed

> Task :core:unitTest FAILED

FAILURE: Build failed with an exception.

SocketServerTest > testClientDisconnectionWithOutstandingReceivesProcessedUntilFailedSend() FAILED
    org.opentest4j.AssertionFailedError: Channel not closed after failed send
        at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:39)
        at org.junit.jupiter.api.Assertions.fail(Assertions.java:117)
        at kafka.network.SocketServerTest.testClientDisconnectionWithOutstandingReceivesProcessedUntilFailedSend(SocketServerTest.scala:1163)
```

[2]
```
❯ ./gradlew clean :core:test --tests kafka.network.SocketServerTest.testClientDisconnectionWithOutstandingReceivesProcessedUntilFailedSend

> Configure project :
Starting build with version 3.1.0-SNAPSHOT using Gradle 7.1.1, Java 11 and Scala 2.13.6

> Task :core:compileScala
Unexpected javac output: warning: [options] bootstrap class path not set in conjunction with -source 8
Note: /Users/ngoel/Workspace/code/apache-kafka/kafka-fork/kafka/core/src/main/scala/kafka/tools/StreamsResetter.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 warning.

> Task :core:compileTestScala
Unexpected javac output: warning: [options] bootstrap class path not set in conjunction with -source 8
1 warning.

> Task :core:test

SocketServerTest > testClientDisconnectionWithOutstandingReceivesProcessedUntilFailedSend() PASSED

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.1.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 3m 52s
84 actionable tasks: 45 executed, 39 up-to-date
```

--- a/core/src/main/scala/kafka/server/ControllerServer.scala
+++ b/core/src/main/scala/kafka/server/ControllerServer.scala
@@ -42,7 +42,7 @@ import org.apache.kafka.raft.RaftConfig
 import org.apache.kafka.raft.RaftConfig.AddressSpec
 import org.apache.kafka.server.authorizer.Authorizer
 import org.apache.kafka.server.common.ApiMessageAndVersion
-import org.apache.zookeeper.server.quorum.QuorumPeerConfig.ConfigException
+import org.apache.kafka.common.config.ConfigException
 
 import scala.jdk.CollectionConverters._
 

